cli/internal/cmd/iamdestroy.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright (c) Edgeless Systems GmbH
+SPDX-License-Identifier: AGPL-3.0-only
+*/
+package cmd
+
+import (
+	"fmt"
+
+	"github.com/edgelesssys/constellation/v2/cli/internal/cloudcmd"
+	"github.com/edgelesssys/constellation/v2/internal/constants"
+	"github.com/edgelesssys/constellation/v2/internal/file"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+func runDestroyIAMUser(cmd *cobra.Command, _args []string) error {
+	spinner := newSpinner(cmd.ErrOrStderr())
+	destroyer := cloudcmd.NewIAMDestroyer(cmd.Context())
+	fsHandler := file.NewHandler(afero.NewOsFs())
+
+	return destroyIAMUser(cmd, spinner, destroyer, fsHandler)
+}
+
+func destroyIAMUser(cmd *cobra.Command, spinner spinnerInterf, destroyer iamDestroyer, fsHandler file.Handler) error {
+	yes, err := cmd.Flags().GetBool("yes")
+	if err != nil {
+		return err
+	}
+
+	if !yes {
+		// Confirmation
+		ok, err := askToConfirm(cmd, "Do you really want to destroy your IAM user?")
+		if err != nil {
+			return err
+		}
+		if !ok {
+			fmt.Println("The destruction of the IAM user was aborted")
+			return nil
+		}
+	}
+
+	if proceed, err := deleteGCPServiceAccountKeyFile(cmd, destroyer, fsHandler); err != nil && !proceed {
+		return err
+	}
+
+	spinner.Start("Destroying IAM User", false)
+
+	if err := destroyer.DestroyIAMUser(cmd.Context()); err != nil {
+		return fmt.Errorf("couldn't destroy IAM User: %w", err)
+	}
```

Comment:
Is it OK to delete the key file before we run `terraform destroy`.
IMO we should do it after to ensure we only delete the file if everything else went according to plan.
@katexochen any opinion on this?

Reply:
This order is kind of needed since the key file is compared with the `sa_key` from the terraform state file to ensure the files belong to the same IAM group. I didn't encounter problems with using this order of deleting files, however, if it seems safer, I could rework this so that the `sa_key` is saved in a variable before the terraform files get deleted.

Reply:
The problem I see is if something goes wrong during `terraform destroy` the key file is already gone.
But since we already check if the key file to be deleted matches the one from the terraform state I guess this is fine.

Reply:
I'd say its fine, too. If something goes wrong during termination we are likely in a state where we cannot continue to use the IAM creds anyway. The only thing that should be possible is to repeat the termination until it succeeds.
Maybe it is even safer this way, I could image a scenario where terraform succeeds to delete the cloud resource of the gcpKey, but the deletion of another resource fails. In this case, the terraform state would be updated and wouldn't contain the gcpKey any longer, but we wouldn't have deleted the local gcpKey file and couldn't match it the next time we run terminate.